flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManagerDefaultResourceAllocationStrategyITCase.java

```diff
@@ -39,7 +39,7 @@
     protected Optional<ResourceAllocationStrategy> getResourceAllocationStrategy() {
         return Optional.of(
                 new DefaultResourceAllocationStrategy(
-                        DEFAULT_TOTAL_RESOURCE_PROFILE, DEFAULT_NUM_SLOTS_PER_WORKER));
+                        DEFAULT_TOTAL_RESOURCE_PROFILE, DEFAULT_NUM_SLOTS_PER_WORKER, false));
```

Comment:
This creates a chance that the evenly evenly spread out configuration is inconsistent between the resource allocation strategy and the slot manager configuration.

I think we should pass the slot manager configuration into `getResourceAllocationStrategy` and read the boolean from it.